lib/xcodeproj/project/object/helpers/file_references_factory.rb

```diff
@@ -237,6 +238,24 @@ def find_products_group_ref(group, should_create = false)
             product_group_ref
           end
 
+          # Find the remote info name for a new sub project.
+          #
+          # @param  [PBXFileReference] ref
+          #         The file reference to compare.
+          #
+          # @param  [Project] project
+          #         The project to which to find the target.
+          #
+          # @note   If the target is found, then return the name of target,
+          #         otherwise, the default value 'Subproject' will be returned
+          #
+          # @return [String] The remote info name.
+          #
+          def find_remote_info_name_for_product_reference(ref, project)
```

Comment:
I think you can re-write this as:

```ruby
          def find_remote_info_name_for_product_reference(ref, project)
            project.native_targets.find { |t| t.product_reference == ref }&.name || 'Subproject'
          end          
```

Reply:
RuboCop complains over this for using the 2.1 parser. Want me to update to 2.3?

Reply:
It complains about the ampersand. Changing to 2.3 complains about frozen string literals and freezing immutable objects and other stuff.